source/GameWindow.cpp

```diff
@@ -46,6 +53,40 @@ namespace {
 
 		return false;
 	}
+
+// The following two functions are only available for Valve Steam Deck support.
+#ifdef __linux__
+	//Files::Read does not work because /sys bytes always returns 4096 for file
+	//size and does not match the contents byte size.
+	string ReadLinuxSysFile(const string &path) {
+		stringstream strStream;
+		ifstream ifs(path);
+		strStream << ifs.rdbuf();
+		string result = strStream.str();
+		// trim trailing newline
+		if (!result.empty() && result[result.length()-1] == '\n') {
+			result.erase(result.length()-1);
+		}
+		return result;
+	}
+
+	// Check if steam deck hardware by reading system vendor and product name.
+	bool IsSteamDeck() {
+		// code name Jupiter is the product for steam deck
+		string DECK_VENDOR = "Valve";
+		string DECK_PRODUCT = "Jupiter";
```

Comment:
Make these `const` and define them outside this function.

Reply:
I'll see what I can do.